test/common/http/http1/header_formatter_test.cc

```diff
@@ -0,0 +1,18 @@
+#include "common/http/http1/header_formatter.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http1 {
+TEST(ProperCaseHeaderKeyFormatterTest, Formatting) {
```

Comment:
Can we significantly bump coverage here including empty values, single character values, etc.